crypto/evp/evp_lib.c

```diff
@@ -387,7 +397,7 @@ int evp_cipher_cache_constants(EVP_CIPHER *cipher)
 
 int EVP_CIPHER_get_block_size(const EVP_CIPHER *cipher)
 {
-    return cipher->block_size;
+    return (cipher == NULL) ? -1 : cipher->block_size;
```

Comment:
Why -1 and not 0?

See my comment in #23000

Reply:
i responded there as well, but for completeness, we have several call checks for expressly < 0 in our code base (as well as other checks that are inconsistent).

If block_size can in fact never be legitimately zero (which seems sensible, given @t8m comment that stream ciphers set a block size of 1), then yeah, this can return 0 instead of -1, but I don't know if thats unilaterally true.

If it is, yeah, I'll make the change, but that will have to come with an audit of the 24-some-odd call sites to ensure that they make the right size check, and arguably an alteration of the definition of the get_block_size functions to return either unsigned int or size_t.

Reply:
I've converted the block_size return code to be 0, and audited the call sites for get_block_size and get_iv_length to ensure they check for the proper range to differentiate errors from valid values.